Segmentation violation with docker build --compress --stream

moved from https://github.com/moby/moby/issues/36579 - just tried, and this still reproduces on Docker 18.03.1


Running `echo "FROM alpine" | docker build --compress --stream . -f -` from an empty (or non-empty) folder produces this error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4554000]

goroutine 50 [running]:
github.com/docker/cli/cli/command/image/build.Compress.func1(0xc42000e040, 0x0, 0x0)
	/go/src/github.com/docker/cli/cli/command/image/build/context.go:389 +0x70
created by github.com/docker/cli/cli/command/image/build.Compress
	/go/src/github.com/docker/cli/cli/command/image/build/context.go:384 +0x102
```

```
docker version
Client:
 Version:	18.02.0-ce-rc1
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	5e1d90a
 Built:	Thu Jan 25 00:33:50 2018
 OS/Arch:	darwin/amd64
 Experimental:	true
 Orchestrator:	kubernetes

Server:
 Engine:
  Version:	18.02.0-ce-rc1
  API version:	1.36 (minimum version 1.12)
  Go version:	go1.9.3
  Git commit:	5e1d90a
  Built:	Thu Jan 25 00:40:43 2018
  OS/Arch:	linux/amd64
  Experimental:	true
```

--- a/cli/command/image/build.go
+++ b/cli/command/image/build.go
@@ -182,6 +182,10 @@ func runBuild(dockerCli command.Cli, options buildOptions) error {
 		remote        string
 	)
 
+	if options.compress && options.stream {
+		return errors.New("--compress conflicts with --stream options")
+	}
+
 	if options.dockerfileFromStdin() {
 		if options.contextFromStdin() {
 			return errors.New("invalid argument: can't use stdin for both build context and dockerfile")

